Providers/Pocket/PocketFeed.swift

```diff
@@ -6,7 +6,24 @@ import Foundation
 import Shared
 import Storage
 
-class Pocket: FeatureFlaggable, URLCaching {
+protocol PocketStoriesProviding {
+    typealias StoryResult = Swift.Result<[PocketFeedStory], Error>
+
+    func fetchStories(items: Int, completion: @escaping (StoryResult) -> Void)
+    func fetchStories(items: Int) async throws -> [PocketFeedStory]
```

Comment:
I moved the `Deferred` implementation to completion based Result type similar with Contiles, as discussed per last PR. But as I was writing the implementation logic for adding the sponsored content, I noticed I would benefit a lot from making the code suspending, instead of completion based hell 😂 so I researched the `asyc/await` pattern there's pretty cool initializer that takes, guess what, a result based completion and converts the function to async. (Really cool!). So because they depend on each other, I wouldn't necessary deprecate the completion method, instead maybe in some cases the completion might be needed. As the time passes, maybe we can rewrite even the `URLSession` code to be async, but at this point we can take some time to learn the benefits/drawback of the new concurrency model. Wdt?

Reply:
Yeah good point, I'm good with that! Thanks for the explanation 😄